cabal-install/Distribution/Client/TargetSelector.hs

```diff
@@ -222,6 +222,8 @@ data TargetString =
    | TargetString2 String String
    | TargetString3 String String String
    | TargetString4 String String String String
+   | TargetString5 String String String String String
+   | TargetString7 String String String String String String String
```

Comment:
Really. You inspired me to have a proper unambigious form for scripts and I realised the existing forms were not necessarily unambigious and were not very consistent or sane. So these are long but they're consistent: each bit gets a qualification, the long forms are:
```
:pkg:foo:lib:foo
:pkg:foo:lib:foo:module:Data.Foo
:pkg:foo:exe:foo:file:Main.hs
```
Count them: 5 and 7 components.